test/integration/submission/submission.spec.ts

```diff
@@ -965,6 +965,143 @@ describe('Submission Api', () => {
+        .catch(err => err);
+    };
+
+    this.beforeEach(async () => {
+      await clearCollections(dburl, ['donors', 'activesubmissions']);
+      donor = await generateDonor(dburl, programId, 'ICGC_0001');
+    });
+    it('should return 401 if no auth is provided', done => {
+      chai
+        .request(app)
+        .post('/submission/program/ABCD-EF/clinical/approve/asdf')
+        .end((err: any, res: any) => {
+          res.should.have.status(401);
+          done();
+        });
+    });
+    it('should return 403 if the user is not DCC Admin', done => {
+      chai
+        .request(app)
+        .post('/submission/program/ABCD-EF/clinical/approve/asdf')
+        .auth(JWT_ABCDEF, { type: 'bearer' })
+        .end((err: any, res: any) => {
+          res.should.have.status(403);
+          done();
+        });
+    });
+    it('should return 404 if no active submission is available', done => {
+      chai
+        .request(app)
+        .post('/submission/program/WRONG-ID/clinical/approve/asdf')
+        .auth(JWT_CLINICALSVCADMIN, { type: 'bearer' })
+        .end((err: any, res: any) => {
+          res.should.have.status(404);
+          done();
+        });
+    });
+    it('should return 400 if an active submission is available with a different version ID', async () => {
+      await uploadSubmission();
+      return chai
+        .request(app)
+        .post(`/submission/program/${programId}/clinical/approve/wrong-version-id`)
+        .auth(JWT_CLINICALSVCADMIN, { type: 'bearer' })
+        .then((res: any) => {
+          res.should.have.status(400);
+        })
+        .catch(err => err);
+    });
+    it('should return 409 if an active submission is available but not in PENDING_APPROVAL state', async () => {
+      await uploadSubmission();
+      await validateSubmission();
+      // State should be approved
+      return chai
+        .request(app)
+        .post(`/submission/program/${programId}/clinical/approve/${submissionVersion}`)
+        .auth(JWT_CLINICALSVCADMIN, { type: 'bearer' })
+        .then((res: any) => {
+          res.should.have.status(409);
+        })
+        .catch(err => err);
+    });
+    it('should return 200 when commit is completed', async () => {
+      // To get submission into correct state (pending approval) we need to already have a completed submission...
+      await uploadSubmission();
+      await validateSubmission();
+      await commitActiveSubmission();
+      // Now we need to have a submission with updates, and validate to get it into the correct state
+      await uploadSubmissionWithUpdates();
+      await validateSubmission();
+      return chai
+        .request(app)
+        .post(`/submission/program/${programId}/clinical/commit/${submissionVersion}`)
+        .auth(JWT_CLINICALSVCADMIN, { type: 'bearer' })
+        .then((res: any) => {
+          res.should.have.status(200);
+          // TODO: check that merge and save were successful
```

Comment:
approving for now but should check if donor is updated properly in DB and activeSubmission is deleted